src/js/jsx/sections/libraries/LibraryBar.jsx

```diff
@@ -63,53 +69,130 @@ define(function (require, exports, module) {
 
         /**
          * Uploads a color asset to the library
-         * @todo Make this accept a color variable instead and correctly provide
-         *       from various sources
          * @private
          */
-        addColorAsset: function () {
-            // FIXME: We're going to need context sensitive boxes for (stroke, fill and overlay color)
-            // For demonstration purposes, this action uses a hard coded color
+        addColorAsset: function (color) {
             // FIXME: We may also need to extend to other color spaces/representations here, check other uses of colors
-            this.getFlux().actions.libraries.createColorAsset({ r: 0, g: 255, b: 128 });
+            this.getFlux().actions.libraries.createColorAsset(color);
+        },
+        
+        /**
+         * Helper function to create color button based on the layer attribute (fills or strokes)
+         * @private
+         */
+        _createColorButton: function (attr, buttonText) {
+            var layer = this._selectedLayers().first();
+            
+            if (this._selectedLayers().size !== 1 || layer[attr].isEmpty()) {
+                return null;
+            }
+            
+            var color = layer[attr].first().color;
+            
+            return (
+                <SplitButtonItem title={buttonText} disabled={this.props.disabled}>
+                    <div className="split-button__item__color-icon"
+                         style={{ backgroundColor: color.toCssRGB() }}
+                         onClick={this.addColorAsset.bind(this, color)} />
+                </SplitButtonItem>
+            );
+        },
+        
+        /**
+         * True if user can add the selected layer(s) as graphic into the libraries.
+         * @private
+         */
+        _canAddGraphic: function () {
+            if (this.props.disabled || this._selectedLayers().isEmpty()) {
+                return false;
+            }
+            
+            var countOfEmptyGroups = 0;
+            
+            this._selectedLayers().forEach(function (layer) {
```

Comment:
Well.. I'm very much against one line wrapper functions that just make the code harder to debug. I'd recommend instead of `_selectedLayers`, just use `document.layers.selected` everywhere.

After that, you can just use `.all()` on your selectedLayers to see if they're all empty groups.